modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinCleanupTest.java

```diff
@@ -257,6 +260,68 @@ public void testToolBarFocus() {
         assertEquals("first item in toolbar must be focused", bar.getItems().get(0), scene.getFocusOwner());
     }
 
+//-------- TabPane
+    @Test
+    public void testChildrenCountAfterSkinIsReplaced() {
+      TabPane tabPane = new TabPane();
+      tabPane.getTabs().addAll(new Tab("0"), new Tab("1"));
+      showControl(tabPane, false);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      TabPaneSkin skin1 = (TabPaneSkin)createAlternativeSkin(tabPane);
+      assertEquals(6, tabPane.getChildrenUnmodifiable().size());
+      tabPane.setSkin(skin1);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+    }
+
+    @Test
+    public void testChildrenCountAfterSkinIsRemoved() {
+      TabPane tabPane = new TabPane();
+      assertEquals(0, tabPane.getChildrenUnmodifiable().size());
+      tabPane.getTabs().addAll(new Tab("0"), new Tab("1"));
+      showControl(tabPane, false);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      root.getChildren().remove(tabPane);
+      assertTrue(tabPane.getSkin() instanceof TabPaneSkin);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      tabPane.setSkin(null);
+      assertNull(tabPane.getSkin());
+      assertEquals(0, tabPane.getChildrenUnmodifiable().size());
+    }
+
+    @Test
+    public void testNPEWhenAddTabsAfterSkinIsReplaced() {
+      TabPane tabPane = new TabPane();
+      tabPane.getTabs().addAll(new Tab("0"), new Tab("1"));
+      showControl(tabPane, false);
+      replaceSkin(tabPane);
+      tabPane.getTabs().addAll(new Tab("2"), new Tab("3"));
+    }
+
+    @Test
+    public void testNPEWhenRemTabAfterSkinIsReplaced() {
+      TabPane tabPane = new TabPane();
```

Comment:
Personally I don't like abbreviation in method names, not even test methods. `testNPEWhenRemoveTabAfterSkinIsReplaced` isn't that much longer :)

Reply:
Corrected.